src/views/core/members/UpdateProfile.vue

```diff
@@ -0,0 +1,217 @@
+              </select>
+            </div>
+          </div>
+          <div class="control">
+            <button class="button is-info" @click="askToJoinBody(tmpBodyId)">Join body</button>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import _ from 'lodash'
+import { mapGetters } from 'vuex'
+
+export default {
+  name: 'UpdateProfile',
+  data () {
+    return {
+      user: {
+        first_name: '',
+        last_name: '',
+        id: null,
+        username: null,
+        primary_body_id: null,
+        primary_body: null,
+        circles: null,
+        date_of_birth: null,
+        gender: null,
+        phone: null,
+        address: null,
+        about_me: null,
+        user: {},
+        bodies: []
+      },
+      bodies: [],
+      tmpBodyId: null,
+      tmpEmail: '',
+      errors: {},
+      birthday: null,
+      isLoading: true,
+      isSaving: false
+    }
+  },
+  methods: {
+    saveUser () {
+      this.isSaving = true
+      this.errors = {}
+
+      const body = _.pick(this.user, [
+        'id',
+        'first_name',
+        'last_name',
+        'username',
+        'gender',
+        'about_me',
+        'address',
+        'phone',
+        'date_of_birth'
+      ])
+
+      this.axios.put(this.services['core'] + '/members/me', body).then(() => {
+        this.isSaving = false
+
+        this.$root.showSuccess('User is saved.')
+
+        /* istanbul ignore next */
+        return this.$router.push({
+          name: 'oms.members.view',
+          params: { id: this.user.username || this.user.id }
+        })
+      }).catch((err) => {
+        this.isSaving = false
+
+        if (err.response.status === 422) { // validation errors
+          this.errors = err.response.data.errors
+          return this.$root.showError('Some of the user data is invalid.')
+        }
+
+        this.$root.showError('Could not save user', err)
+      })
+    },
+    askToJoinBody (bodyId) {
+      this.$buefy.dialog.prompt({
+        message: 'Join body',
+        inputAttrs: {
+          placeholder: 'Motivation',
+          required: false
+        },
+        onConfirm: (motivation) => this.joinBody(bodyId, motivation)
+      })
+    },
+    joinBody (bodyId, motivation) {
+      this.isLoading = true
+      this.axios.post(this.services['core'] + '/bodies/' + bodyId + '/join-requests', {
+        motivation
+      }).then(() => {
+        this.$root.showSuccess('Join request is sent.')
+        this.isLoading = false
+        this.isRequestingMembership = true
+      }).catch((err) => {
+        this.isLoading = false
+        if (err.response.status === 422) {
+          this.$root.showError('You\'ve already requested to join this body.')
+        } else {
+          this.$root.showError('Could not sent join request', err)
+        }
+      })
+    },
+    changeEmail (newEmail) {
+      if (['aegee.eu', 'aegee.org'].some(domain => newEmail.includes(domain))) {
```

Comment:
Maybe reference this back to the constant set in the other file? Instead of copying the values